content/blog/jt15s/2020-12-31_hello_from_the_haiku_promotion_team.md

```diff
@@ -0,0 +1,74 @@
+
+### General Users
+We mustn't forget about what is probably our main user group: general users. General users of Haiku are those who come to explore and try out the OS. Due to Haiku's remaining rough edges and still lurking bugs (we are in Beta after all), our general users are typically more understanding of the limitations and issues with the project. Attracting more general users and making sure our general userbase is content is very important.
+Strategies such as making sure user queries are answered as well as spreading the word about new apps and releases is a good first start, but we are considering various other methods too.
+
+### App Developers
+If you follow technology news, you might have seen a mention of Medo, the recently-released 4K editor clocking in at only 1.44MB! This number alone highlights one of Haiku's many development advantages: our apps are small and light - even if you have a project with lots of features crammed into it, on Haiku the file size will likely be very reasonable. Add that to the fact that we have HaikuPorts, an organised porting system, that we use C++, one of the most popular progamming languages in the world, and that we are compatible with frameworks like Qt, a popular UI frontend framework, and you get the development environment that is Haiku. We want to attract more app developers, whether that's developers who want to port their existing app over to Haiku, or new developers who wish to make an app in Haiku that they feel is needed.
+Additionally, we want to make any existing ports of FOSS projects to Haiku official platforms - this means having Haiku listed as an officially supported platform on the project's website. By getting Haiku listed on projects such as KDE, more people will be aware of Haiku and the fact that Haiku has a multitude of apps available to download.
+
+### Researchers
+Surprisingly, Haiku is also an excellent system for research purposes. Let me explain further. Researchers don't need the stability and reliability that businesses do, which means Haiku is a possible research system for them. Furthermore, it is possible to make invasive and disruptive changes to Haiku without having to dodge security restrictions imposed by the system, or having to "hack" their way into the system. In the past, Haiku has had successful collaborations with the Auckland University, where a group working on UI/UX research experimented with Haiku. 
+They produced Stack&Tile as well as the Auckland Layout Engine (ALE) for laying out user interfaces using a constraint solver. The layout system in Apple's iOS today is based on this research work. Apart from ALE, the group also had other projects which were not implemented into Haiku, such as a tool to auto-generate user documentation from user interface code as well as a way to dynamically modify the layout of a running window to optimise it for a specific workflow.
```

Comment:
It would be nice to have links here that prove claims such as that Apple's iOS layout system is based on work done by the Auckland University on Haiku.

Reply:
The documentation from Apple: https://developer.apple.com/library/archive/documentation/UserExperience/Conceptual/AutolayoutPG/index.html

I don't think it credits the Auckland University explicitly, but the concepts are very similar.

Actually it predates the Auckland Layout Editor (Apple had this since 2011 and the paper from Auckland university was published in 2013: https://www.researchgate.net/publication/262402078_The_Auckland_Layout_Editor_An_improved_GUI_layout_specification_process)

I'm not sure how later versions of the layout system at Apple got ideas from that paper, as they kept adding things in later versions of their system (for example a summary of their changes is here: https://sarunw.com/posts/history-of-auto-layout-constraints/ )

Reply:
Since we don't know if Apple based their system on the ALE or not, I've just removed the line saying that. I've also added a link to the research paper too.